### ISSUE TITLE
command line options seem ignored

### ISSUE THREAD

**Issue by tko** [rust-lang/rust-clippy#5706]
```rust
const REF_U: f64 = 0.19783000664283680764;
```
Running `cargo clippy` the code above triggers `excessive_precision` warning:
```
warning: float has excessive precision
[..]
  = note: `#[warn(clippy::excessive_precision)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#excessive_precision
```
`cargo clippy -- --allow excessive_precision` _also_ triggers a warning instead of allowing it and being quiet about it.
Same for `--deny excessive_precision` and `--forbid excessive_precision`

clippy 0.0.212 (d4092ace 2020-05-11)

**Comment by matthiaskrgr** [comment]
If it's a clippy lint, you need to prefix the lint name with `clippy::`.

`RUSTFLAGS="-Aclippy::excessive_precision" cargo clippy"` does not issue a warning here.

**Comment by tko** [comment]
It is a clippy lint. I missed the need for prefix, thanks for pointing it out. However using `clippy::excessive_precision` makes no difference when using the command line options as described by `cargo clippy --help`
```
To allow or deny a lint from the command line you can use `cargo clippy --`
with:

    -W --warn OPT       Set lint warnings
    -A --allow OPT      Set lint allowed
    -D --deny OPT       Set lint denied
    -F --forbid OPT     Set lint forbidden
```
RUSTFLAGS as in your example seems to work.

**Comment by flip1995** [comment]
It works for me with `clippy 0.0.212 (0262de5 2020-06-07)` and `cargo clippy -- -Aclippy::excessive_precision` and `cargo clippy -- --allow clippy::excessive_precision`

**Comment by flip1995** [comment]
Also with `clippy 0.0.212 (99cb9cc 2020-05-11)`.

Can you try a `cargo clean` before running `cargo clippy -- -Aclippy::excessive_precision`, please?

**Comment by tko** [comment]
uh, yes,  `cargo clean` does it.

It seems it's not so much that the command line arguments are ignored but rather they're _persisted_.
```sh
cargo clean
cargo clippy -- -Aclippy::excessive_precision
cargo clippy
```
Logically the second `cargo clippy` should be showing warnings, but for me it is silent, as if `-A` was used. I can see it the other way round too, seems the flags or lack thereof from the first run are used on the second run.

Apparently `touch src/lib.rs` between clippy runs also makes the second one run with the new flags.

**Comment by flip1995** [comment]
Ah ok, this is tracked in https://github.com/rust-lang/rust-clippy/issues/4612#issuecomment-618725621
